mutiny-core/src/nostr/nwc.rs

```diff
@@ -999,6 +999,11 @@ impl NostrWalletConnect {
                                             nostr_manager.logger,
                                             "Payment timeout, not removing payment from budget"
                                         );
+
+                                        nostr_manager.save_nwc_profile(self.clone())?;
+
+                                        // don't save to pending list, we already paid it
+                                        return Ok(None);
```

Comment:
Payments that timeout, but then fail, will not be included in the list for them to retry. That's a huge miss.

Reply:
good catch